### ISSUE TITLE
DaprClient.GetMetadataAsync() throwing Exception

### ISSUE THREAD

**Issue by sanjeev-saxena-us** [dapr/dotnet-sdk#1349]
Dapr.Client .NET package, this used to work in 1.10 but appears to be broken in 1.12 onwards; is now throwing a "Object reference not set to an instance". Have tried with 1.14 and same issue.

**Comment by alicejgibbons** [comment]
Able to reproduce with the following findings:
- only happens when `app-port` (app channel) is set and hasn't been found/established by daprd
- Running without app channel or when daprd has discovered the app already allows `DaprClient.GetMetadataAsync()` to work fine.

Eg. Running a sample dotnet app like the following works fine, regardless of where the `GetMetadataAsync()` command is
`dapr run --app-id DaprClient --dapr-http-port 5780 --dapr-grpc-port 5701 -- dotnet run`

But running
`dapr run --app-id DaprClient --app-port 5292 --dapr-http-port 5780 --dapr-grpc-port 5701` fails with the following error when calling `GetMetadataAsync()` ***before*** the application has been discovered on specified port. 

```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Dapr.Client.DaprClientGrpc.GetMetadataAsync(CancellationToken cancellationToken)
   at Program.<Main>$(String[] args) in /Users/alicegibbons/Documents/Customers/Alegeus/DaprClientTest/TodoApi/Program.cs:line 15
   at Program.<Main>(String[] args)
```

**Comment by JoshVanL** [comment]
Based on my limited DotNet experience, I believe the issue here is that the [DotNet client method](https://github.com/dapr/dotnet-sdk/blob/3d1fa01d0f52298c880f8d92b08690aa317be270/src/Dapr.Client/DaprClientGrpc.cs#L2394) is not checking for a nil pointer of the gRPC message for `ActorRuntime`. It is currently the case that `ActorRuntime` in the Metadata message will be nil until the app port is available and read by daprd.

```
&runtime.GetMetadataResponse{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc0003d95d0)}, sizeCache:0, unknownFields:[]uint8(nil), Id:"cd4386a7-c513-4d72-bd2c-c808e6f63e55", ActiveActorsCount:[]*runtime.ActiveActorsCount(nil), RegisteredComponents:[]*runtime.RegisteredComponents(nil), ExtendedMetadata:map[string]string{"daprRuntimeVersion":"edge"}, Subscriptions:[]*runtime.PubsubSubscription(nil), HttpEndpoints:[]*runtime.MetadataHTTPEndpoint(nil), AppConnectionProperties:(*runtime.AppConnectionProperties)(0xc00088d860), RuntimeVersion:"edge", EnabledFeatures:[]string(nil), ActorRuntime:(*runtime.ActorRuntime)(nil)}
```

```
&runtime.GetMetadataResponse{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(0xc0003d95d0)}, sizeCache:0, unknownFields:[]uint8(nil), Id:"cd4386a7-c513-4d72-bd2c-c808e6f63e55", ActiveActorsCount:[]*runtime.ActiveActorsCount(nil), RegisteredComponents:[]*runtime.RegisteredComponents(nil), ExtendedMetadata:map[string]string{"daprRuntimeVersion":"edge"}, Subscriptions:[]*runtime.PubsubSubscription(nil), HttpEndpoints:[]*runtime.MetadataHTTPEndpoint(nil), AppConnectionProperties:(*runtime.AppConnectionProperties)(0xc00088de00), RuntimeVersion:"edge", EnabledFeatures:[]string(nil), ActorRuntime:(*runtime.ActorRuntime)(0xc00088de60)}
```

```
ActorRuntime:(*runtime.ActorRuntime)(nil)}
```

```
ActorRuntime:(*runtime.ActorRuntime)(0xc00088de60)}
```

**Comment by marcduiker** [comment]
Same issue as #1296

**Comment by WhitWaldo** [comment]
@marcduiker Thanks for finding the duplicate!

@JoshVanL Your conclusion looks accurate to me. We have a lot of opportunities for .NET client improvement here:
- There's no unit test coverage for the `GetMetadataAsync` method or return type
- Lack of nullability annotations

I've created a PR with a fix for this issue and a pending PR (#1317 ) that adds nullability annotations, so we can hopefully keep a few more of these from slipping through the cracks soon.
